src/reducers/__tests__/subCategorizeAll.ts

```diff
@@ -89,28 +90,42 @@ it('set cursor on new empty thought', () => {
 
 })
 
-it('move all visible and hidden thoughts into a new empty thought after subcategorizeAll', () => {
+it('move all non meta thoughts and only allowed meta thoughts into new empty thought after subCategorizeAll', () => {
+
+  const metaExceptions = SUBCATEGORIZE_ALL_META_EXCEPTIONS.reduce(
+    (acc, meta, index) => (acc + `    - ${meta}${index === (SUBCATEGORIZE_ALL_META_EXCEPTIONS.length - 1) ? '' : '\n'}`)
+  , '')
```

Comment:
I would hardcode the exceptions into the test. We want tests to be as concrete as possible. Principles of abstraction that we use in application architecture do not apply in the test environment. It forces each case to be considered individually, which is what we want to do if there are new meta attributes.

Of course, we do generalize tests at times but we have to be careful to avoid complexity and abstraction when possible.

Reply:
Understood.